src/mjolnir/shortcutbuilder.cc

```diff
@@ -471,6 +457,15 @@ uint32_t AddShortcutEdges(GraphReader& reader,
       bool diff_names = directededge->endnode().tileid() == edge_id.tileid() &&
                         !OpposingEdgeInfoMatches(tile, directededge);
 
+      // Recalculate speed limit to include turn durations
+      uint32_t average_speed = edgeinfo.speed_limit();
+      if (edgeinfo.speed_limit() != 0 && total_turn_duration > 0) {
+        // meters per second // TODO is there a converter? remove zeros
+        auto const speed_mps = edgeinfo.speed_limit() * 1000.0 / 3600;
+        auto const actual_time = length / speed_mps + total_turn_duration;
+        average_speed = static_cast<uint32_t>(std::round(length / actual_time * 3600 / 1000));
```

Comment:
i wouldnt call it average speed. its not an average

Reply:
Naming is difficult 😞. "tuns_aware_speed" is also clumsy